### PR TITLE
CB-25. Java 11 support.

### PR DESCRIPTION
**Pull request by meowfaceman**
Java 11 is now supported in Cloudbreak. This additionally modifies
several docker images to user openjdk 11 instead of openjdk 10.

### CODE REVIEW

**Comment by meowfaceman** [comment]
I'll note that I had to change the test-image Dockerfile as I don't have access to the hwx_openjdk. I think the test-image is different enough from the official cloudbreak image to warrant its own Dockerfile, but I'll defer to your expertise here.

**Comment by meowfaceman** [comment]
Some noteworthy things:

- Gradle 5 and spring-boot 2.1 are both required for Java 11, which precipitate a bunch of other dependency changes, including a new springframework version and a new asm version.
- Gradle 5 made exclusion syntax more explicit, annoyingly.
- GuavaCache was deprecated in spring in favor of Caffeine, which was created as a replacement for GuavaCache.
- AutoCloseable was removed from micrometer (https://github.com/micrometer-metrics/micrometer/issues/757).
- spring bean overriding was made non-default in spring-boot 2.1, which broke some tests. This is why the spring property is being passed in manually.
- A few references to TestNG collections were modified to point to Guava collections instead.
- TestNG itself runs into a bug that was fixed in the 7.0.0 line.
- The test-image in integration tests was made non-reliant on the hortonworks/cloudbreak image. This makes it easier for me to run integration tests, but otherwise I'm not married to this approach.

**Comment by mhmxs** [comment]
Please do not merge, we are waiting for infra changes

**Comment by meowfaceman** [comment]
I'll hold off.

File: integration-test/test-image/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+
+WORKDIR /
+
+# Install zip
+RUN apt-get update --no-install-recommends && apt-get install -y zip procps wget && apt-get clean && rm -rf /var/lib/apt/lists/*
 
 ADD cloudbreak.jar /
+ADD start_cloudbreak_app.sh /
+ADD wait_for_cloudbreak_api.sh /
+ADD jmx_prometheus_javaagent-0.10.jar /
```
Comment (daszabo) [inline integration-test/test-image/Dockerfile]: We are using Micrometer, bundled with Spring Boot. I am not sure if this jar is still needed

**Comment by keyki** [comment]
Are we moving the ball for this?

**Comment by meowfaceman** [comment]
AFAIK, still waiting on infra changes.

**Comment by doktoric** [comment]
@meowfaceman can we help in infra changes?

**Comment by meowfaceman** [comment]
@mhmxs Is the one who did the infra changes as I didn't have access to the Hortonworks Jenkins at the time. I believe it's waiting on the addition of Java 11 to various Jenkins build nodes.

**Comment by mhmxs** [comment]
This issue is still blockerd

File: integration-test/test-image/Dockerfile (hunk 1 above)
Comment (meowfaceman) [reply integration-test/test-image/Dockerfile]: Just noticed this comment now -- in the interests of not changing too much with this change, I'm going to leave this as is and we can file a separate JIRA for removing the jar -- this JAR is still used to generate the official Cloudbreak container as well.

**Pull request merged**